src/nbl/video/IOpenGL_LogicalDevice.h

```diff
@@ -792,17 +788,17 @@ class IOpenGL_LogicalDevice : public ILogicalDevice, protected impl::IOpenGL_Log
                 _waitAll = true;
                 timeout = 0xdeadbeefBADC0FFEull;
             }
-            while (true)
+            for (bool notFirstRun=false; true; notFirstRun=true)
             {
                 for (uint32_t i=0u; i<_count; )
                 {
-                    const auto now = clock_t::now();
+                    const auto now = SRequestWaitForFences::clock_t::now();
                     if (timeout)
                     {
-                        if(now>=end)
+                        if(notFirstRun && now>=timeoutPoint)
```

Comment:
maybe first `now` should be taken from outside (as param, queried from clock before submitting the request)? with very low timeouts it's possible it'll always be timeout without any GL call

Reply:
`notFirstRun` condition should enforce at least one call to `glClientSyncWait`

Reply:
`notFirstRun` condition should enforce at least one call to `glClientSyncWait`